File: 剑指Offer/Java/用两个栈实现队列.java
import java.util.Stack;
 
public class Solution {
    Stack<Integer> stack1 = new Stack<Integer>();
    Stack<Integer> stack2 = new Stack<Integer>();
 
    public void push(int node) {
        stack1.push(node);
    }
 
    public int pop() {
        while (stack1.isEmpty() == false){
            stack2.push(stack1.pop());
        }
        int val = stack2.pop();
        while (stack2.isEmpty() == false){
            stack1.push(stack2.pop());
        }
        return val;
    }
}

File: 剑指Offer/C++/用两个栈实现队列.cpp
class Solution
{
public:
    void push(int node) {
        stack1.push(node);
    }

    int pop() {
        int res;
        while (!stack1.empty()){
            int data = stack1.top();
            stack2.push(data);
            stack1.pop();
        }
        res = stack2.top();
        stack2.pop();
        while (!stack2.empty()){
            int data = stack2.top();
            stack1.push(data);
            stack2.pop();
        }
        return res;
    }

private:
    stack<int> stack1;
    stack<int> stack2;
};